presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/KeyProperty.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
```

Comment:
Same comment as above re. move to spi

Reply:
see response to your previous comment

> Same comment as above re. move to spi

Please see response to your previous comment.